packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -463,6 +586,87 @@ export function normalizeAllowedTypes(
 	return normalized;
 }
 
+/**
+ * Normalizes an allowed type to an {@link AnnotatedAllowedType}, by adding empty annotations if they don't already exist.
+ *
+ * @internal
+ */
+export function normalizeToAnnotatedAllowedType<T extends LazyItem<TreeNodeSchema>>(
+	type: T | AnnotatedAllowedType<T>,
+): AnnotatedAllowedType<T> {
+	return isAnnotatedAllowedType(type)
+		? type
+		: {
+				metadata: {},
+				type,
+			};
+}
+
+/**
+ * Converts an {@link ImplicitAnnotatedAllowedTypes} to an {@link ImplicitAllowedTypes}s, by removing
+ * any annotations.
+ *
+ * @internal
+ */
+export function unannotateImplicitAllowedTypes<Types extends ImplicitAnnotatedAllowedTypes>(
```

Comment:
This file as several new functions: its units tests should gain sections for these functions with at least a minimal unit test each.